tests/ui/never_type/const-context.rs

```diff
@@ -0,0 +1,7 @@
+// build-pass
+
+trait X<const N: i32> {}
+
+fn hello<T: X<{ fn hello() -> ! { loop {} } 1 }>>() {}
```

Comment:
could you rename this? it's not really that it's in a "const context", it's more about it being in a nested item.

Maybe `never-type-in-nested-fn-decl`

Reply:
:+1: